unitTests/ui/CategoryFacet/CategoryFacetTest.ts

```diff
@@ -412,5 +412,45 @@ export function CategoryFacetTest() {
         expect(test.env.queryController.executeQuery).not.toHaveBeenCalled();
       });
     });
+
+    describe("when there's many parent values to display", () => {
+      beforeEach(() => {
+        simulateQueryData = buildSimulateQueryData(30, 30);
+        Simulate.query(test.env, simulateQueryData);
+      });
+
+      it('should only return the non-ellipsed values as visible', () => {
+        const visibles = test.cmp.getVisibleParentValues();
+
+        // NUMBER_OF_VALUES_TO_KEEP_AFTER_TRUNCATING parents + 1 child value
+        expect(visibles.length).toBe(CategoryFacet.NUMBER_OF_VALUES_TO_KEEP_AFTER_TRUNCATING + 1);
+      });
+
+      it('should properly build the ellipse boundaries', () => {
+        const visibles = test.cmp.getVisibleParentValues();
+
+        // Check first value
+        expect(visibles[0].value).toContain('parent0');
+
+        // Check value right before ellipse
```

Comment:
Instead of the comment, it would be better to use one or two well-named variables. e.g.

```
const indexBeforeEllipse = CategoryFacet.NUMBER_OF_VALUES_TO_KEEP_AFTER_TRUNCATING / 2 - 1;
const valueBeforeEllipse = visibles[indexBeforeEllipse].value;
expect(valueBeforeEllipse).toContain('parent25');
```

I would apply this to all the expectations.

It would also be good to convert this test into a `describe` block with 5 `it` blocks, one for each aspect. This would make it easier to pin-point errors should something fail.